go.mod

```diff
@@ -74,6 +74,7 @@ require (
 	go.opentelemetry.io/otel/sdk v1.10.0
 	go.opentelemetry.io/otel/trace v1.10.0
 	go.opentelemetry.io/proto/otlp v0.19.0
+	go.uber.org/goleak v1.2.0
```

Comment:
I see this is much better but current method is simple and "good enough" for our purposes, do we want to pull a dependency just for this?  I defer to deps people to judge this

@liggitt @dims

Reply:
I wouldn't want to add a dependency for this, especially not one that has any transitive dependencies

Reply:
> current method is simple and "good enough" for our purposes, 

It's not :disappointed: 

The current code doesn't support ignoring the goroutines that were started during init. That's crucial for reliably ignoring some of the goroutines that currently force us to the fuzzy "3 goroutine leaks are okay" thing.

> We do have some code we can reorganize into a package if needed:

I had started improving that code, but then realized that I would end up re-implementing goleak.

Is the problem with goleak the dependencies of the package? The actual code that we pull is pretty simple.

What about simply copying that code into `test/utils/goleak`?

Reply:
`check-dependency-stats` doesn't look too bad to me:

```
Switched to a new branch 'base'
--- /logs/artifacts/stats-base.txt	2023-02-01 16:00:51.332458491 +0000
+++ /logs/artifacts/stats.txt	2023-02-01 16:00:50.944422049 +0000
@@ -1,6 +1,6 @@
 Direct Dependencies: 210 
-Transitive Dependencies: 235 
-Total Dependencies: 285 
+Transitive Dependencies: 236 
+Total Dependencies: 286 
 Max Depth Of Dependencies: 23 
 All dependencies:
 
@@ -249,6 +249,7 @@
 go.uber.org/multierr
 go.uber.org/zap
 golang.org/x/crypto
+golang.org/x/lint
 golang.org/x/mod
 golang.org/x/net
 golang.org/x/oauth2
```

Reply:
we have https://github.com/kubernetes/kubernetes/tree/master/third_party/forked

Reply:
The integration tests are now passing with the vague "allow 3 additional goroutines" replaced with more precise suppressions through `goleak.Ignore*`. I found that in two cases I needed `goleak.IgnoreTopFunction`. I'd prefer just using goleak as-is instead of investing further effort into improving our own code or forking goleak (regardless whether that's under "third_party/forked" or "test").

@liggitt: WDYT, is `golang.org/x/lint` as new additional dependency besides goleak itself acceptable?

Or is this about other dependencies that don't show up in `go.mod` and `check-dependency-stats`?

Reply:
I opened https://github.com/uber-go/goleak/issues/92 with a request / suggestion for dropping the golang.org/x/lint reference from the graph

Reply:
I submitted a PR for goleak. Let's wait for that.